From 3c5ee489df3b989c4b8d54cb976c61fd8f72dace Mon Sep 17 00:00:00 2001
From: Dave Sugar <dsugar100@gmail.com>
Date: Tue, 18 Mar 2025 14:40:47 -0400
Subject: [PATCH 39/43] Fix mislabeling of /etc/shadow

/etc/shadow was getting labeled shadow_history_t due to the fact
that when /usr/sbin/unix_update (labeled updpwd_exec_t) creates
the temporary /etc/nshadow, it was getting shadow_history_t.
When /etc/nshadow was renamed to /etc/shadow, shadow_history_t
was retained, causing an issue.

This change is causing updpwd_t to label /etc/nshadow as shadow_t
so when moved, /etc/shadow has the correct type.

Signed-off-by: Dave Sugar <dsugar100@gmail.com>
---
 policy/modules/system/authlogin.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/authlogin.te b/policy/modules/system/authlogin.te
index 6c775ee61..f671c0e9e 100644
--- a/policy/modules/system/authlogin.te
+++ b/policy/modules/system/authlogin.te
@@ -392,6 +392,7 @@ allow updpwd_t self:fifo_file rw_fifo_file_perms;
 allow updpwd_t self:unix_stream_socket create_stream_socket_perms;
 allow updpwd_t self:unix_dgram_socket create_socket_perms;
 
+files_etc_filetrans(updpwd_t, shadow_t, file, "nshadow")
 files_etc_filetrans(updpwd_t, shadow_history_t, file)
 allow updpwd_t shadow_history_t:file manage_file_perms;
 
-- 
2.49.0

